src/main/java/org/identifiers/cloud/ws/resolver/api/controllers/ResolverApiController.java

```diff
@@ -105,6 +115,33 @@ public ResponseEntity<?> resolve(@PathVariable String resolutionRequest, HttpSer
         return new ResponseEntity<>(result, result.getHttpStatus());
     }
 
+    @GetMapping(value = "/resolveMirId/{mirId}")
+    public ResponseEntity<?> resolve(@PathVariable String mirId) {
+        try {
+            URI namespaceLocation = model.resolveMirId(mirId);
+            if (namespaceLocation != null) {
+                HttpHeaders headers = new HttpHeaders();
+                headers.setLocation(namespaceLocation);
+                return new ResponseEntity<>(headers, HttpStatus.FOUND);
+            } else {
+                return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+            }
+        } catch(NumberFormatException e) {
+            ServiceResponse response = new ServiceResponse();
+            response.setApiVersion(ApiCentral.apiVersion);
+            response.setErrorMessage("MIRIDs must be in the format MIR:XXXXXXXX or integers up to 8 digit");
+            return new ResponseEntity<>(response, HttpStatus.BAD_REQUEST);
+        } catch (URISyntaxException e) {
+            // Should never actually happen on production.
+            // If so, check the WS_RESOLVER_CONFIG_REGISTRY_NAMESPACE_REDIRECT_FORMAT application property
+            ServiceResponse response = new ServiceResponse();
+            response.setApiVersion(ApiCentral.apiVersion);
+            response.setErrorMessage("MIRID resolution format is not setup correctly on server. Please contact support.");
```

Comment:
Is this message going to be displayed to users (on an error screen)? If so, it might be useful to have a link to the support form, or the support email address included in the message.

Reply:
Good point